src/batchUploader.ts

```diff
@@ -31,6 +37,9 @@ export class BatchUploader {
     mpInstance: MParticleWebSDK;
     uploadUrl: string;
     batchingEnabled: boolean;
+    private eventVault: BaseVault<SDKEvent[]>;
+    private batchVault: BaseVault<Batch[]>;
```

Comment:
We're simply defining that `eventVault` and `batchVault` can be of type `BaseVault`, which can be either a `SessionStorageVault` or `LocalStorageVault`. We're doing the same thing two lines down for uploader on line 43.

Reply:
I thought that you shouldn't ever reference an abstract class unless you are using it to extend a class, which was the reason for my comment. But maybe I'm wrong about that?

Reply:
My thinking is usually to err on the side of being more explicit, so if you know what kind of base class you're dealing with, use that. If all you know is that is it will be _some_ base class, then thats fine. 

If `eventVault` here will always be a `SessionStorageVault`, then yes lets go with that, but if it could be any type of vault at this stage, then it would be correct to identify it as just a `BaseVault`